### PR TITLE
M3-1137 Create Tag (Linodes Detail)

### PR DESCRIPTION
**Pull request by martinmckenna**
### Purpose

Create a Tag on Linodes Detail

### Notes
* First and foremost, `<LinodesDetail />` got a facelift. Much of the logic was abstracted to the `<LinodesDetailHeader />` component, as to get the line numbers down and readability up
* Also fixed an issue where deleting a Tag on Linodes Detail would delete the Tag permanently, when really it should have been removing that tag from the current Linode
* Incorporated `react-select`. There was a bit of magic involved to get it to behave the way I wanted. Please let me know if anything is unclear so I can document it better
   * basically, if you add or delete a Tag, the auto-suggestions will be updated to either add or filter out the Tag you added or removed to the suggestion list

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+                 * which holds the list of tags queued for deletion 
+                 */
+                return eachTag === inProgressTag;
+              })}
+            />
+          )
+        })}
+        {(this.state.isCreatingTag)
+          ? <Select
+            onChange={this.handleCreateTag}
```
Comment (Jskobos) [inline src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx]: indentation

File: src/services/linodes.ts [hunk 2]
```diff
@@ -380,3 +380,12 @@ export const startMutation = (linodeID: number) => {
   )
     .then(response => response.data)
 }
+
+export const addTagsToLinode = (linodeID: number, tags: string[]) => {
```
Comment (Jskobos) [inline src/services/linodes.ts]: Wouldn't an updateLinode call with { tags } do the same thing without an extra method?

File: src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+            options={this.state.tagsToSuggest}
+            variant='creatable'
+            errorText={this.state.tagError}
+            onBlur={this.handleToggleCreate}
```
Comment (Jskobos) [inline src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx]: This logic needs to be tweaked. If I click the button but then don't focus the Select, it's hard to exit the creation state. We don't really have a final design for this whole business, but maybe a cancel button or something would be helpful?

Comment (Jskobos) [reply src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx]: Oh, or we could manually pop open the select when the plus is clicked.

File: src/services/linodes.ts (hunk 2 above)
Comment (rjstires) [reply src/services/linodes.ts]: Yes, agreed, this is an unnecessary abstraction in the services lib. Lets keep these as generic and as close to matching the API as possible.

File: src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx (hunk 3 above)
Comment (rjstires) [reply src/features/linodes/LinodesDetail/LinodesDetailHeader.tsx]: Agreed with both points. It feels like we're trying to shoehorn a select menu in here TBH. I'd argue that a small popover modal would be a more efficient way to handle tags.

**Comment by martinmckenna** [comment]
@rjstires Let me know what you think about the changes. Instead of making `<LinodesDetailHeader />` into a presentation component, I instead created smaller components.

imo, it makes sense to have logic in `<LinodesDetailHeader />` because most of the functions in that component rely on `linode` not being `undefined` which is definitely a possibility in `<LinodesDetail />`. Further, if all that logic lived in `<LinodesDetail />`, it would be massive.
